scripts/build_lilypond.sh

```diff
@@ -14,18 +14,29 @@ LILYPOND_SRC="$LILYPOND/src"
 LILYPOND_BUILD="$LILYPOND/build"
 LILYPOND_INSTALL="$LILYPOND/install"
 
+if [[ $VERBOSE -ne "0" ]]; then
+    echo "Environment variables:"
+    echo "LILYPOND=$LILYPOND"
+    echo "LILYPOND_SRC=$LILYPOND_SRC"
+    echo "LILYPOND_BUILD=$LILYPOND_BUILD"
+    echo "LILYPOND_INSTALL=$LILYPOND_INSTALL"
+    echo ""
+fi
+
 echo "Extracting '$LILYPOND_TAR'..."
 mkdir -p "$LILYPOND_SRC"
 tar -x -f "$LILYPOND_TAR" -C "$LILYPOND_SRC" --strip-components 1
 
-echo "Building lilypond..."
+echo "Building LilyPond..."
 mkdir -p "$LILYPOND_BUILD"
 (
     cd "$LILYPOND_BUILD"
 
     # Load shared srfi modules.
     export LD_LIBRARY_PATH="$GUILE_INSTALL/lib${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}"
-    export LDFLAGS="-Wl,--export-dynamic"
+    if [ $(uname) != "Darwin" ]; then
+        export LDFLAGS="-Wl,--export-dynamic"
+    fi
```

Comment:
Does this work on macOS without the flag? I got errors when it tries to load the srfi modules

Reply:
Yes, it does. FYI, here's the configuration I use for Guile v1.8.8 in my own Lily dev setup:

```
CFLAGS="-Wno-deprecated-declarations -Wno-misleading-indentation -Wno-unused-but-set-variable -I$(brew --prefix readline)/include -I$(brew --prefix libtool)/include -I$(brew --prefix gmp)/include" \
LDFLAGS="-L$(brew --prefix readline)/lib -L$(brew --prefix libtool)/lib -L$(brew --prefix gmp)/lib" \
CC=gcc-9 \
./configure --prefix=~/src/lily-deps/install
make && make check && make install
```

Reply:
Yes, but I think this means you're building dynamic libraries. Then libguile.so provides the needed symbols for srfi. In the scripts I'm avoiding this as much as possible, so the binary has to export the symbols that come from the static library.

Reply:
Correct, I'm building dynamic libs. Come to think of it, I removed this because `clang` complained about this option. I'll try with `gcc`.

Reply:
Note: This is very much WIP, and ATM I've just thrown a lot of things in here. I'll clean it more up before I push again.

Reply:
No, this is passed to the linker. If it throws errors, it might be that the one on macOS doesn't support this option. You can try to get through all the scripts without the option, but it might not work in the end. In that case you need to find a replacement.

Reply:
Yes, I know it belongs to the linker, and what you say is correct.

Reply:
Seems like we need this on macOS (from `man ld`):
```
 -export_dynamic
    Preserves all global symbols in main executables during
    LTO.  Without this option, Link Time Optimization is
    allowed to inline and remove global functions. This option
    is used when a main executable may load a plug-in which
    requires certain symbols from the main executable.
```

Changed to `-Wl,-export_dynamic` for LP and Guile builds.